src/Sarif/EnumeratedArtifact.cs

```diff
@@ -64,6 +66,39 @@ public byte[] Bytes
             set => this.bytes = value;
         }
 
+        public string CleanPath
+        {
+            get => GetCleanFilePath(Uri.OriginalString);
+            set => this.cleanPath = value;
+        }
+
+        private readonly static Regex AbsoluteFilePathWithFileSchemeRegex = new Regex(@"file:(?:\/{3}|\/{2}\w+\/|\/{1})?(.+)", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        private string GetCleanFilePath(string originalString)
+        {
+            if (this.cleanPath != string.Empty)
+            {
+                return this.cleanPath;
+            }
+
+            if (originalString.StartsWith("file:"))
+            {
+                Match match = AbsoluteFilePathWithFileSchemeRegex.Match(originalString);
+                if (match.Success)
+                {
+                    string separator = Path.DirectorySeparatorChar.ToString();
+                    this.cleanPath = match.Groups[1].Value.Replace("/", separator);
+                    return this.cleanPath;
+                }
+
+                throw new IOException("Invalid file path format.");
```

Comment:
This doesn't look like an IO exception, no IO was performed to get to it.
 
Also, CleanPath shouldn't be property if the getter can throw this.